sparse/client.go

```diff
@@ -39,34 +26,40 @@ func SyncFile(localPath string, addr TCPEndPoint, remotePath string, timeout int
 	for retries := 1; retries >= 0; retries-- {
 		hashLocal, err = syncFile(localPath, addr, remotePath, timeout, retries > 0)
 		if err != nil {
-			if _, ok := err.(*HashCollsisionError); ok {
-				// retry on HahsCollisionError
-				log.Warn("SSync: retrying on chunk hash collision...")
-				continue
-			} else {
-				log.Error("SSync error:", err)
-			}
+			log.Error("SSync error:", err)
 		}
 		break
 	}
 	return
 }
 
 func syncFile(localPath string, addr TCPEndPoint, remotePath string, timeout int, retry bool) ([]byte, error) {
-	file, err := fio.OpenFile(localPath, os.O_RDONLY, 0)
+	file, err := os.OpenFile(localPath, os.O_RDONLY, 0)
 	if err != nil {
 		log.Error("Failed to open local source file:", localPath)
 		return nil, err
 	}
+	log.Debugf("opened file: %s", localPath)
+
 	defer file.Close()
 
-	size, err := file.Seek(0, os.SEEK_END)
+	fileSize, err := file.Seek(0, os.SEEK_END)
 	if err != nil {
 		log.Error("Failed to get size of local source file:", localPath, err)
 		return nil, err
 	}
 
-	SetupFileIO(size%Blocks == 0)
+	directIO := (fileSize%Blocks == 0)
+	SetupFileIO(directIO)
```

Comment:
Will this function called outside of package? If not then it should be lower case in the first character for private function.

Also, it should use interface as I commented in `SetupFileIO`